Mathlib/Analysis/Complex/UpperHalfPlane/Basic.lean

```diff
@@ -353,21 +353,24 @@ theorem neg_smul (g : GL(2, ℝ)⁺) (z : ℍ) : -g • z = g • z := by
   simp only [num, denom, Complex.ofReal_neg, neg_mul, GLPos.coe_neg_GL, Units.val_neg, neg_apply]
   ring_nf
 
+lemma denom_one (τ : ℍ) : denom 1 τ = 1 := by
+  simp [denom]
+
 section SLModularAction
 
-namespace ModularGroup
+open ModularGroup
 
 variable (g : SL(2, ℤ)) (z : ℍ) (Γ : Subgroup SL(2, ℤ))
 
 @[simp]
-theorem sl_moeb (A : SL(2, ℤ)) (z : ℍ) : A • z = (A : GL(2, ℝ)⁺) • z :=
+theorem ModularGroup.sl_moeb (A : SL(2, ℤ)) (z : ℍ) : A • z = (A : GL(2, ℝ)⁺) • z :=
```

Comment:
This is still inside the `UpperHalfPlane` namespace, so the full name of this lemma will still be `UpperHalfPlane.ModularGroup.sl_moeb`; the duplication problem isn't solved.

I suggest collecting all lemmas / defs which take an element of SL(2, Z) as an argument together at the end of the file, so you can close the `UpperHalfPlane` namespace and open the `ModularGroup` namespace. This will also help to make the file more readable, it's a bit confusing as it is now.